src/tests/test_dal.py

```diff
@@ -30,3 +30,21 @@ def __init__(self, resource_manager=None):
     # Verify self._dal is set to the optional DataAccessLayer class i.e. TestDal class
     dm = TestDataManager()
     assert isinstance(dm._dal, TestDal)
+
+
+def test_dal_getitem_access():
+    """
+    Verify `__getitem__` is identical to `__getattr__`
+    """
+
+    expected = "foo"
+
+    class SampleService(Service):
+        def sample_method(self):
+            return expected
+
+    dm = _BaseDataManager()
+    dm.register_services(sample=SampleService())
+
+    with dm.context() as ctx:
+        assert ctx.dal['sample.sample_method']() == expected
```

Comment:
Would be good to confirm that middleware ran. The original implementation worked, it would return `expected`, but it would not have ran middleware.